mesonbuild/options.py

```diff
@@ -93,16 +97,23 @@ class OptionKey:
     __slots__ = ['name', 'subproject', 'machine', '_hash']
 
     name: str
-    subproject: str
+    subproject: T.Optional[str] # None is global, empty string means top level project
     machine: MachineChoice
     _hash: int
 
-    def __init__(self, name: str, subproject: str = '',
+    def __init__(self,
+                 name: str,
+                 subproject: T.Optional[str] = None,
                  machine: MachineChoice = MachineChoice.HOST):
+        if not isinstance(machine, MachineChoice):
+            raise MesonException(f'Internal error, bad machine type: {machine}')
+        if not isinstance(name, str):
+            raise MesonException(f'Key opane is not a string: {name}')
```

Comment:
If we are concerned that there's a coding error in the implementation and want to add guards against that, we should raise MesonBugException instead.

Reply:
Ok but as I said -- code that is an extra security measure shouldn't be using MesonException, which prints out:

```
ERROR: Key opane is not a string: XXXX
```

Instead, we want to raise MesonBugException:

```
ERROR: key opane is not a string: XXXX

    This is a Meson bug and should be reported!
```

This helps in two ways:
- if users experiment with the merged PR and hit this error, it is immediately apparent to them that they did nothing wrong and meson just has a bug. Instead of spending hours reading documentation and playing with their build files trying to get the error to go away, they simply click "submit new bug" and go on their merry way.
- it aids reviewers who say things like "we shouldn't need this", because MesonBugException tells said reviewers that this is simply a guardrail for code that has not been fully exercised.